__tests__/json_decode_test.ml

```diff
@@ -1,6 +1,42 @@
 open Jest
 open Expect
 
+module Test = struct
+  type default_case = 
+    | Float
+    | Int
+    | String
+    | Null
+    | Array
+    | Object
+    | Bool
+  
+  (* TODO: tests for this function *)
+  let test decoder = 
+    let open Json in function
+    | Float -> test "float" (fun () ->
+        expectFn decoder (Encode.float 1.23) |> toThrow)
+    | Int -> test "int" (fun () ->
+      expectFn decoder (Encode.int 23) |> toThrow);
+    | String -> test "string" (fun () ->
+      expectFn decoder (Encode.string "test") |> toThrow);
+    | Null -> test "null" (fun () ->
+      expectFn decoder Encode.null |> toThrow);
+    | Array -> test "array" (fun () ->
+      expectFn decoder (Encode.array [||]) |> toThrow);
+    | Object -> test "object" (fun () ->
+      expectFn decoder (Encode.object_ @@ Js.Dict.empty ()) |> toThrow);
+    | Bool -> test "boolean" (fun () ->
+      expectFn decoder (Encode.boolean Js.true_) |> toThrow);
+  ;;
+
+  let rec throws decoder = function
+    | [] -> ();
+    | default_case::t ->
```

Comment:
This naming confuses me. I don't see what's default about `default_case`, and I think a `t` without a corresponding `h` is confusing. I'd prefer it to be spelled out as `tail`, unless it's in the more conventional form of `h::t` (and `head::tail`, or `first::rest`, would be even better, usually)